backend/src/main/scala/cromwell/backend/standard/callcaching/StandardCacheHitCopyingActor.scala

```diff
@@ -100,28 +101,50 @@ abstract class StandardCacheHitCopyingActor(val standardParams: StandardCacheHit
       } yield (callOutputs, destinationDetritus, simpletonCopyPairs ++ detritusCopyPairs)
 
       processed match {
-        case Success((callOutputs, destinationDetritus, allCopyPairs)) =>
-          duplicate(allCopyPairs) match {
+        case Success((callOutputs, destinationDetritus, allIoCommands)) =>
+          duplicate(ioCommandsToCopyPairs(allIoCommands)) match {
             case Some(Success(_)) => succeedAndStop(returnCode, callOutputs, destinationDetritus)
             case Some(Failure(failure)) => failAndStop(failure)
             case None =>
-              val allCopyCommands = allCopyPairs map { case (source, destination) => copyCommand(source, destination, overwrite = true) }
-
-              allCopyCommands foreach { sendIoCommand(_) }
 
-              goto(WaitingForCopyResponses) using Option(StandardCacheHitCopyingActorData(allCopyCommands, callOutputs, destinationDetritus, returnCode))
+              if (allIoCommands.nonEmpty) {
+                allIoCommands foreach { sendIoCommand(_) }
+                
+                goto(WaitingForIoResponses) using Option(StandardCacheHitCopyingActorData(allIoCommands, callOutputs, destinationDetritus, returnCode))
+              } else succeedAndStop(returnCode, callOutputs, destinationDetritus)
           }
 
         case Failure(failure) => failAndStop(failure)
       }
   }
 
-  when(WaitingForCopyResponses) {
-    case Event(IoSuccess(copyCommand: IoCopyCommand, _), Some(data)) =>
-      val newData = data.remove(copyCommand)
-      if (newData.copyCommandsToWaitFor.isEmpty) succeedAndStop(data.returnCode, data.copiedJobOutputs, data.copiedDetritus)
+  when(WaitingForIoResponses) {
+    case Event(IoSuccess(command: IoCommand[_], _), Some(data)) =>
+      val newData = data.remove(command)
+      if (newData.commandsToWaitFor.isEmpty) {
+        onSuccessIoCommand(data) match {
```

Comment:
I agree that the non-`None` pattern might be a bit clunky. 
But fusing this one with the rest of the commands would be even harder to follow IMO as it would look like "send all those commands but not this last one, instead wait for all the others to come back successfully and then send that one or if it failed do that instead".
Which is kind of what this is doing but using a separate FSM state which I think makes it clearer ?